contracts/settling_game/modules/relics/Relics.cairo

```diff
@@ -97,31 +110,118 @@ func set_relic_holder{pedersen_ptr: HashBuiltin*, syscall_ptr: felt*, range_chec
+    );
 
-    let (holds_relic) = uint256_eq(winners_relic_owner, loser_token_id);
+    let (winner_realm_data: RealmData) = IRealms.fetch_realm_data(realms_address, winner_token_id);
 
-    if (holds_relic == TRUE) {
-        _set_relic_holder(winner_token_id, winner_token_id);
+    let (loser_relics_len) = owned_relics_count.read(loser_token_id);
+
+    loop_claim_order_relic(0, realms_address, winner_realm_data, loser_token_id, loser_relics_len);
+
+    return ();
+}
+
+func loop_claim_order_relic{pedersen_ptr: HashBuiltin*, syscall_ptr: felt*, range_check_ptr}(
+    index: felt,
+    realms_address: felt,
+    winner_realm_data: RealmData,
+    loser_token_id: Uint256,
+    loser_relics_len: felt,
+) {
+    if (index == loser_relics_len) {
         return ();
     }
 
-    // TODO: Add Order capture back
+    let (relic_id) = owned_relics.read(loser_token_id, index);
+
+    let (relic_owner_data: RealmData) = IRealms.fetch_realm_data(realms_address, relic_id);
+
+    if (winner_realm_data.order == relic_owner_data.order) {
+        _set_relic_holder(relic_id, relic_id);
+        return ();
+    }
+    // _set_relic_holder(relic_id, relic_id);
+
+    loop_claim_order_relic(index + 1, realms_address, winner_realm_data, loser_token_id, loser_relics_len);
+
+    return ();
+}
+
+@external
+func return_relics{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+    realms_token_id: Uint256
+) {
+    // Only Settling
+    Module.only_approved();
+    let (relic_count) = owned_relics_count.read(realms_token_id);
+
+    loop_return_relics(0, realms_token_id, relic_count);
+
+    return ();
+}
+
+func loop_return_relics{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+    index: felt, realms_token_id: Uint256, relic_count: felt
+) {
+    if (index == relic_count) {
+        return ();
+    }
+
+    let (relic_id) = owned_relics.read(realms_token_id, index);
+
+    _set_relic_holder(relic_id, relic_id);
+
+    loop_return_relics(index + 1, realms_token_id, relic_count);
 
     return ();
 }
 
+func get_relic_index{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+    relic_id: Uint256, owner_token_id: Uint256
+) -> (index: felt) {
+    let (relics_len: felt, relics: Uint256*) = get_owned_relics(owner_token_id);
+
+    let (index) = Relics._get_relic_index(0, relics_len, relics, relic_id);
+
+    return (index=index);
+}
+
 // -----------------------------------
 // SETTERS
 // -----------------------------------
 
 func _set_relic_holder{pedersen_ptr: HashBuiltin*, syscall_ptr: felt*, range_check_ptr}(
     relic_id: Uint256, owner_token_id: Uint256
 ) {
+    alloc_locals;
+    // Get old relic holder, then update array
+    let (old_owner_token_id) = storage_relic_holder.read(relic_id);
+
     storage_relic_holder.write(relic_id, owner_token_id);
 
+    // If old owner exists, update array
+    let (check_exists) = uint256_lt(Uint256(0,0), old_owner_token_id);
```

Comment:
This will never be true, will it?

Reply:
It will be true when the owner has previously been set from a battle.

Example (-> = wins battle, Realm A and C same order):

Realm B -> Realm A (A's relic set to realm B, `storage_relic_holder(A) = B`)
Realm C -> Realm B (A's relic returned, B's relics updated to 0. `storage_relic_holder(A) = B` not uint(0))

Reply:
Ah, my bad, I read it wrong. Isn't the issue tho, contrary to my previous message, that it will always be true?

In the simulation you describe above, why is B's relic set to 0? Doesn't C capture it? Where in the code does this setting to 0 happen? Is it the `owned_relics.write(old_owner_token_id, relic_index, Uint256(0, 0));` call 2 lines below? If so, why is it a different storage_var?

Sorry, it looks like I'm not aware of some piece of here 🥶

Reply:
C won't capture it because A and C are in the same order, so that relic needs to be given back to A rather than captured. After those 2 battles:

A has relic A.
B has no relics.
C has relics C and B.

Yes every index in the `owned_relics` array of for B needs to be set to 0. B was previously holding 2, it's own and A's, we are updating the array to keep track that it no longer holds A's. This array storage is needed because B could have n number of other realms relics.

Reply:
`let (check_exists) = uint256_lt(Uint256(0,0), old_owner_token_id);` will be false at the first loss of each realm. There is no initialization of relic owners, so when `storage_relic_holder = uint(0)` then the realm holds it own relic and hasn't lost it yet

Reply:
The `old_owner_token_id` is never going to be 0 (*), is it not? You're setting zeros in the `owned_relics` storage var, but comparing for zero a val read from `storage_relic_holder`. And the only place `storage_relic_holder` is being written to is in this function, with the new `owner_token_id` value.

(*) -- actually, **initally** the whole `storage_relic_holder` is going to be zero, because all the values are uninitialized. 

> every index in the owned_relics array of for B needs to be set to 0

You're only setting a single index tho, not the whole array.

Reply:
the `loop_claim_order_relic` calls `_set_relic_holder` for each relic, so we set the single index but for each relic for the all relics B owns (in the case above)